Update warnings regarding workspace identity modifications (#2952)
MicrosoftDocs/fabric-docs: docs/security/workspace-identity.md

> [!WARNING]
> Modifying or deleting the service principal or app registration in Azure is not recommended, as it will cause Fabric items relying on workspace identity to stop working. Such changes may be reverted.

> > Adding API permissions to allow access to target resources is supported and shouldn't cause the workspace identity to stop working.

Reviewer: ```suggestion
> Adding API permissions to allow access to target resources is supported for workspace identities.
```

> > Modifications to add API permissions to allow access to a target resources are safe to do and should not cause the workspace identity to stop working.

Reviewer 2: ```suggestion
> Adding API permissions to allow access to target resources is supported and shouldn't cause the workspace identity to stop working.
```

> [!WARNING]
> Modifications to the application made here will cause the workspace identity to stop working, and such changes may be reverted.

> > Modifications to add API permissions to allow access to a target resources are safe to do and should not cause the workspace identity to stop working.

Reviewer 2: ```suggestion
> Adding API permissions to allow access to target resources is supported and shouldn't cause the workspace identity to stop working.
```
